cmd/launcher/flare.go

```diff
@@ -2,55 +2,87 @@ package main
 
 import (
 	"context"
+	"flag"
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
 
 	"github.com/go-kit/kit/log"
-	"github.com/kolide/kit/env"
 	"github.com/kolide/kit/ulid"
 	"github.com/kolide/launcher/pkg/agent/flags"
 	"github.com/kolide/launcher/pkg/agent/knapsack"
+	"github.com/kolide/launcher/pkg/agent/storage/inmemory"
 	"github.com/kolide/launcher/pkg/debug/checkups"
+	"github.com/kolide/launcher/pkg/debug/shipper"
 	"github.com/kolide/launcher/pkg/launcher"
 )
 
+// runFlare is a command that runs the flare checkup and saves the results locally or uploads them to a server.
 func runFlare(args []string) error {
 	// Flare assumes a launcher installation (at least partially) exists
 	// Overriding some of the default values allows options to be parsed making this assumption
 	// TODO this stuff needs some deeper thinking
 	launcher.DefaultAutoupdate = true
-	setDefaultPaths()
-	_ = launcher.DefaultBinDirectoryPath
-
-	opts, err := launcher.ParseOptions("flare", args)
-	if err != nil {
-		return err
-	}
+	launcher.SetDefaultPaths()
 
 	var (
-		dirPath = env.String("KOLIDE_AGENT_FLARE_ZIP_DIR_PATH", "")
+		flagset = flag.NewFlagSet("flare", flag.ExitOnError)
+		flSave  = flagset.String(
+			"save",
+			"local",
+			"local | upload",
+		)
+		flOutputDir = flagset.String(
+			"output_dir",
+			".",
+			"path to directory to save flare output",
+		)
+		flUploadRequestURL = flagset.String(
+			"upload_request_url",
+			"",
+			"URL to request a signed upload URL",
+		)
 	)
 
-	id := ulid.New()
-	reportName := fmt.Sprintf("kolide_agent_flare_report_%s", id)
-	reportPath := fmt.Sprintf("%s.zip", filepath.Join(dirPath, reportName))
+	if err := flagset.Parse(args); err != nil {
```

Comment:
go back to using ff?

Reply:
`ff` has two things that the stdlib doesn't -- it'll parse the environment for values, and it'll read a config file. I don't think either matter here. (But `ff` is a light enough wrapper it doesn't really matter)